es/es-server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -515,10 +516,18 @@ protected final GetResult getFromSearcher(Get get, BiFunction<String, SearcherSc
             }
         }
 
+        final DocIdAndSeqNo docIdAndSeqNo;
+        try {
+            docIdAndSeqNo = VersionsAndSeqNoResolver.loadDocIdAndSeqNo(searcher.reader(), get.uid());
```

Comment:
In later ES versions they've integrated the `seqNo` and `primaryTerm` into the `DocIdAndVersion` class, see https://github.com/elastic/elasticsearch/blob/d31207a0a50a76b52aa2233f33c49a5a2e0ab2d4/server/src/main/java/org/elasticsearch/index/engine/Engine.java#L624-L631

That looks neater to me. Would it be much effort to backport these changes into this PR?